### PR TITLE
refactor-timer

### PR DESCRIPTION
**Pull request by sirpolomotro**
Changed timer to use native observer pattern instead of using its own.

### CODE REVIEW

File: core/src/cg/group4/game_logic/stroll/events/StrollEvent.java [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
+ * Interface that gets implemented by every event.
+ *
+ * @author Nick Cleintuar
+ * @author Benjamin Los
+ * @author Martijn Gribnau
+ */
+public abstract class StrollEvent implements Disposable, Observer {
+
+    /**
+     *
```
Comment (foresterre) [inline core/src/cg/group4/game_logic/stroll/events/StrollEvent.java]: Needs javadoc. (empty javadoc)

File: core/src/cg/group4/game_logic/stroll/events/TestStrollEvent.java [hunk 2]
```diff
@@ -1,283 +1,287 @@
...
+     */
+    protected static final int AMOUNT_OF_TASKS = 6;
+    /**
+     * Number of tasks that the player must complete before the event is considered a success.
+     */
+    protected static final int REWARDS = 10;
+    /**
+     * Constants used for each task case.
+     */
+    protected static final int MOVE_LEFT = 1, MOVE_RIGHT = 2, MOVE_DOWN = 3, MOVE_UP = 4,
```
Comment (foresterre) [inline core/src/cg/group4/game_logic/stroll/events/TestStrollEvent.java]: Might want to reformat this to one constant per line (for readability).

File: core/test/cg/group4/GdxTestRunner.java [hunk 3]
```diff
@@ -24,73 +24,72 @@
...
+        synchronized (invokeInRender) {
+            //add for invoking in render phase, where gl context is available
+            invokeInRender.put(method, notifier);
+        }
+        //wait until that test was invoked
+        waitUntilInvokedInRenderMethod();
+    }
+
+    /**
+     *
```
Comment (foresterre) [inline core/test/cg/group4/GdxTestRunner.java]: Empty javadoc.

File: core/test/cg/group4/LauncherTest.java [hunk 4]
```diff
@@ -1,9 +1,5 @@
 package cg.group4;
 
-import org.junit.Test;
-
-import static org.junit.Assert.assertTrue;
-
 public class LauncherTest {
```
Comment (foresterre) [inline core/test/cg/group4/LauncherTest.java]: Empty test class. Might want to remove this.

**Comment by foresterre** [comment]
A few minor comment issues. The code is looking good (Including the observers we spoke about earlier). I will put it tomorrow through inCode; maybe it will find something I missed.

File: core/test/cg/group4/stroll/StrollTest.java [hunk 5]
```diff
@@ -1,35 +0,0 @@
...
-    @Before
-    public void setUp() {
-        TimeKeeper.getInstance().init();
-        cStroll = new Stroll();
-        TimeKeeper.getInstance().getTimer("STROLL").unsubscribe(cStroll.cTimerTask);
-    }
-
-    @Test
-    public void testGeneratePossibleEvent() {
-        assertTrue(true);
```
Comment (Surskit) [inline core/test/cg/group4/stroll/StrollTest.java]: Useless test

File: core/test/cg/group4/stroll/StrollTest.java [hunk 6]
```diff
@@ -1,35 +0,0 @@
...
-    }
-
-
-
-    @After
-    public void tearDown(){
```
Comment (Surskit) [inline core/test/cg/group4/stroll/StrollTest.java]: tearDown preferably above the tests, under the setUp

**Pull request merged**